tests/test_printer.py

```diff
@@ -399,10 +398,44 @@ def test_print_block_without_arguments():
     assert io.getvalue() == """%0 : i32, %1 : i32\n  "test.op"(%1) : (i32) -> ()"""
 
 
+def test_print_block_with_terminator():
+    """Print a block and with its terminator."""
+    block = Block(ops=[test.TestOp.create(), test.TestTermOp.create()])
+
+    # Print block ops including block terminator
+    io = StringIO()
+    p = Printer(stream=io)
+    p.print_block(block, print_block_terminator=True)
+    assert (
+        io.getvalue()
+        == """
+^0:
+  "test.op"() : () -> ()
+  "test.termop"() : () -> ()"""
+    )
+
+
+def test_print_block_without_terminator():
+    """Print a block and its terminator separately."""
+    term_op = test.TestTermOp.create()
+    block = Block(ops=[test.TestOp.create(), term_op])
+
+    # Print block ops separately from the block terminator
+    io = StringIO()
+    p = Printer(stream=io)
+    p.print_block(block, print_block_terminator=False)
+    p.print("\n  ")
+    p.print_op(term_op)
+    assert (
+        io.getvalue()
+        == """\n^0:\n  "test.op"() : () -> ()\n  "test.termop"() : () -> ()"""
```

Comment:
Can you make this into a multi line string as well?

Reply:
Also, I think you could remove lines 427 and 428 and trim this test a bit to make it clearer what you are actually testing?

Reply:
I'm trying to print the block without its terminator and the terminator separately, so I have to add the newlines manually; Not sure how else to express this, it is similar to the block args tests here.

Reply:
Why do you need to print the terminator separately for the test? Isn't it enough to show that the terminator is not printed when you set the flag?

Reply:
We don't actually, I was just copying what was happening with the block args tests in this test file, in case I was missing anything.
I'll remove it.